### DISCUSSION TITLE
Force people to give feedback (from a preset list) to their match when they unmatched.

### DISCUSSION THREAD

**Discussion by WilliamDormer** [PullDating/Org#86] [Ideas & Features]
This goes hand in hand with the idea of limiting the number of concurrent matches that a user can have, but when they unmatch someone it should first ask if they went out on a date with them or not. Secondly, it should give them a pre-set list of feedback that they can choose from for what went wrong. This would then be shown to the other party so that they can see what they did and improve. Additionally, there would be an option for when people decide they are going steady to give them a little celebration and disable their account.

**Comment by ajbott** [comment]
I don't think users would like to submit the surveys as much if the other person knows where they came from and would cause the other person to feel uncomfortable giving honest feedback. Instead, having someone in the middle to analyze the information and relay it to the user would be a good middle ground in gathering enough information on what's been going wrong in their dating life and being able to offer feedback on their situation.

**Reply by WilliamDormer** [reply]
Not saying we should necessarily do this, but feeding in written replies from multiple people into a language model would produce some general amalgamation of the different opinions, anonymizing the feedback a bit but still delivering the message.

**Comment by flo-kei** [comment]
I like this idea since it makes people more aware of the choices they make. (e.g. rejecting s.o. based on x / y)
However I would not force this onto people but rather encourage them to use it, since it helps both them and others.
Also, it should be clearly stated that the given feedback does not show up on the other users profile directly. 
In my opinion **Important**: Make sure that having only very few matches does not break the system (e.g. user B sees he generally messes up something particular and has only matched User A so far, therefore knows, User A specified this as their reason to end the connection.) 
It should only be shown to the User, after they have already ended a couple of connections.

**Comment by ajbott** [comment]
There are a lot of details to hash out to make this system work and make everyone feel comfortable providing feedback, but feedback can be extremely valuable when it comes to dating experience. I think everyone wants to improve, and feedback on why people want to end the conversation, aren't interested, or generally what went wrong gives people more information to decide which direction they want to take.

**Comment by Oshyan** [comment]
This is definitely a problem I want to try to solve. We should probably separately consider 1: *how* to implement a feedback/final message ("unmatch message") feature and 2: how to incentivize using it, and whether to make it mandatory.

As far as how to implement it, I see it as a prompt shown when you unmatch someone which shows a small set of predefined/suggested options (e.g. "Met someone else", "Not feeling a connection", etc.) and an option for "Other/Personalize" to write a short parting message. 

For incentivizing, I do see this as being an optional thing for people to do, but it would be good to think of ways we could encourage people to use it. Perhaps it could become part of a person's "response rating" or something. This is a feature we may not have discussed much yet (or at all), but I do think is worth also considering, e.g. a sort of score that aggregates response time, whether a response was given to all messages/percentage of messages (note: different than "intros"), etc. Curious if people have other ideas for making people *want* to provide a parting message...

**Reply by WilliamDormer** [reply]
I like the idea of having some sort of reputation score to ward off ghosting, late replies, very short replies, not leaving feedback etc...
"+ rep" lol

**Comment by ajbott** [comment]
# Feature: Unmatch Feedback System

## Description:
Implement a feature that prompts users to provide feedback when they choose to unmatch with someone on the app. This feedback system aims to improve user experience and foster constructive communication between users.

## Key Components:
1. **Unmatch Prompt:** When a user decides to unmatch with another user, a prompt will appear asking if they went on a date with the person.
2. **Pre-set Feedback Options:** Users will be presented with a set of predefined options to choose from, indicating reasons for unmatching (e.g., "Met someone else", "Not feeling a connection", etc.).
3. **Personalized Feedback:** Additionally, users can select the "Other/Personalize" option to write a short custom message explaining their decision.
4. **Feedback Delivery:** The selected feedback will be anonymously shared with the unmatched user to provide insight into why the connection ended.
5. **Incentives:** Consider implementing incentives to encourage users to provide feedback, such as incorporating it into a reputation score or response rating system.

## Benefits:
1. **Enhanced User Experience:** Provides users with a platform to express their reasons for unmatching, promoting transparency and closure in interactions.
2. **Improved Communication:** Facilitates constructive communication between users by offering insights into why connections end.
3. **Encourages Engagement:** Incentivizes users to participate in the feedback process, contributing to a more active and involved community.

## Implementation:
1. **UI Integration:** Integrate the unmatch feedback prompt seamlessly into the user interface, ensuring ease of use and accessibility.
2. **Backend Support:** Develop backend functionality to store and deliver feedback data anonymously between users.
3. **Incentive System:** Design and implement a system to reward users for providing feedback, potentially through a reputation or response rating mechanism.
4. **Testing and Iteration:** Conduct thorough testing to ensure the feature operates smoothly and effectively, gathering user feedback for further refinement.

## Conclusion:
The Unmatch Feedback System serves as a valuable tool for promoting transparency, communication, and user engagement on the app. By providing users with the opportunity to share their feedback when unmatching, the feature contributes to a more positive and constructive dating experience for all users.

**Reply by ajbott** [reply]
The key idea is to provide users with feedback on their interactions while ensuring anonymity. We want the feedback to be generalized based on the majority opinion, making it difficult for users to guess who said what. This approach aims to offer constructive insights without revealing individual identities, fostering a more open and comfortable environment for sharing feedback.

**Reply by Oshyan** [reply]
Interesting. I understand the general desire for anonymity, avoiding revenge stuff, etc. That said I envisioned the feature partly if not entirely being *non*-anonymous. Something like a "parting message" from a known match. The user that is unmatched with would not be able to reply, nor view the sender's profile anymore, but would know who sent it from their list of prior matches. It'd be difficult for them *not* to know even if it's technically anonymized, at least not without some fancy wizardry, delaying the feedback until some time after they unmatch or something. I'm in favor of avoiding unnecessary complexity.

Perhaps "Make my feedback anonymous" could be an *option* on the feedback screen but not the default. But I think allowing and even encouraging non-anonymous, *considerate* feedback would be most helpful for the user being "rejected". Knowing who is "rejecting" us is in itself helpful information, and if that comes with some explanation that is specific to that person that is much more helpful than "Some matches feel you talk too much about yourself" or something that's not clearly identified with a person.

Yes, I know there are *potential* safety risks. To that I'd say a few things. First, in most cases someone using an unmatch probably hasn't given the other person enough info to track them down. If they have, and they're motivated to do so, then they're probably not going to be stopped by mere anonymizing of feedback. They would have had to be collecting info on the person already, and would know who unmatched them even without a parting message. Granted a parting message might make them angrier (maybe), but it's not what's making them *dangerous*. Finally there is the point that providing feedback would be optional (albeit encouraged), so for users that do have a heightened sense of danger (whether due to real experience or simply greater concern), they can always choose not to use it.

So my preferred initial implementation, subject to ongoing user feedback, would be:
1. Optional non-anonymous feedback
2. Add "Anonymize" option later if there is sufficient user demand, because it increases complexity

Also just to say in general, I am in favor of doing things that genuinely increase safety, and I also recognize the importance of the *feeling* of safety, even if that depends in part on something that is not rationally a safety issue. Making users feel safe is vitally important and could be tricky. But I also don't want to simply agree to every limit or safety feature that any user asks for or wants. In the process of trying to meet every user's desire, we could easily end up reducing a lot of the potential value and features of the system for the majority. In the end there *is* some inherent risk in using a dating app, and I think any dating app publisher needs to find and draw the line themselves on what risks they are asking of their users. The clearer and more aware and compassionate we can be about that, the better.

**Comment by ajbott** [comment]
## Feature Request: Enhanced Feedback System

**Description:**
Implement an enhanced feedback system to gather and present user feedback in a manner that ensures anonymity and reliability. This system aims to provide users with valuable insights into their interactions while safeguarding their privacy.

**Key Components:**
1. **Aggregated Feedback:** Aggregate user feedback to ensure anonymity and reliability. Only display feedback that has been referenced by a significant number of users, such as 6 or more respondents.
   
2. **Incremental Updates:** Present feedback updates incrementally to maintain anonymity. Consider revealing updates in batches, such as every 10 responses, rather than immediately after each submission.

**Benefits:**
- Enhances user experience by providing valuable insights into interactions.
- Protects user privacy by ensuring anonymity in feedback presentation.
- Encourages user engagement and participation in providing feedback.

**Implementation Considerations:**
- Develop a robust backend system to aggregate and analyze user feedback.
- Design a user-friendly interface to present feedback updates seamlessly.
- Implement privacy measures to safeguard user anonymity throughout the feedback process.

**Conclusion:**
The enhanced feedback system will contribute to a positive user experience by offering valuable insights while prioritizing user privacy. By implementing this feature, we aim to foster a supportive and constructive environment for user interactions within the app.

**Reply by ajbott** [reply]
This feature proposes an advanced feedback mechanism designed to collect and present user feedback anonymously and effectively. By aggregating feedback and implementing incremental updates, it ensures user privacy while offering valuable insights into user interactions. This feature promises to enhance user experience and engagement within the app.
